create/application.go

```diff
@@ -102,10 +103,21 @@ func (app *applicationCmd) Run(ctx context.Context, client *api.Client) error {
 		if err := auth.Valid(); err != nil {
 			return fmt.Errorf("the credentials are given but they are empty: %w", err)
 		}
-		// for git auth we create a separate secret and then reference it in the app.
+
 		secret := auth.Secret(newApp)
+		// for git auth we create a separate secret and then reference it in the app.
 		if err := client.Create(ctx, secret); err != nil {
-			return fmt.Errorf("unable to create git auth secret: %w", err)
+			if k8sError.IsAlreadyExists(err) {
+				// only update the secret if it is managed by nctl in the first place
+				if _, exists := newApp.Annotations[util.ManagedByAnnotation]; exists {
+					fmt.Println("updating git auth credentials")
+					if err := client.Update(ctx, secret); err != nil {
```

Comment:
Not 100% sure on this but can you just call `Update` without getting the resource first? See `update/application.go` where we already do this. Also might be worth adding a test here.